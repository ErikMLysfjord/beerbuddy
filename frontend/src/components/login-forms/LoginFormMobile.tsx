import { Card, theme } from "antd";
import { ValidateErrorEntity } from "rc-field-form/lib/interface";
import HopsUpLogo from "../../assets/hopsup.svg";
import styles from "./LoginFormMobile.module.css";
import LoginFormMobileContainer from "./LoginFormMobileContainer";
import RegisterFormMobileContainer from "./RegisterFormMobileContainer";

const { useToken } = theme;

interface LoginFormMobileProps {
  showMessage: (string: { username: string; password: string }) => void;
  onFinishFailed: (errorInfo: ValidateErrorEntity) => void;
}

//chose last index of pathname to determine if login or signup
const login = window.location.pathname.split("/")[2] === "login";

const LoginFormMobile = ({
  showMessage,
  onFinishFailed,
}: LoginFormMobileProps) => {
  const { token } = useToken();
  return (
    <main className={styles.mobileContainer}>
      <a className={styles.logo} href="/" aria-label="HopsUp">
        <img src={HopsUpLogo} alt="HopsUp-logo" />
        <span>HopsUp</span>
      </a>

      <Card
        style={{
          backgroundColor: token.colorPrimaryBg,
        }}
        className={styles.mobileCard}
      >
        {login ? (
          <LoginFormMobileContainer
            showMessage={showMessage}
            onFinishFailed={onFinishFailed}
          />
        ) : (
          <RegisterFormMobileContainer
            showMessage={showMessage}
            onFinishFailed={onFinishFailed}
          />
        )}
        <a
          style={{ color: token.colorPrimary }}
          className={styles.loginLink}
          href={"/project2/" + (login ? "signup" : "login")}
          tabIndex={0}
        >
          {login ? "Not a user? Sign up here" : "Already a user? Log in here"}
        </a>
      </Card>
    </main>
  );
};

export default LoginFormMobile;
