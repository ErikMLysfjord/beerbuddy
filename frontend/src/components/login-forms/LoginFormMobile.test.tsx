import { describe, it, vi, expect } from "vitest";
import { render } from "@testing-library/react";
import LoginFormMobile from "./LoginFormMobile";
import { axe } from "jest-axe";

const showMessage = vi.fn();
const onFinishFailed = vi.fn();
const saveUser = vi.fn();

describe("LoginFormMobile", () => {
  it("is accessible", async () => {
    const { container } = render(
      <LoginFormMobile
        showMessage={showMessage}
        onFinishFailed={onFinishFailed}
        saveUser={saveUser}
      />
    );

    expect(await axe(container)).toHaveNoViolations();
  });

  it("matches snapshot", async () => {
    const { container } = render(
      <LoginFormMobile
        showMessage={showMessage}
        onFinishFailed={onFinishFailed}
        saveUser={saveUser}
      />
    );
    expect(container).toMatchSnapshot();
  });
});
