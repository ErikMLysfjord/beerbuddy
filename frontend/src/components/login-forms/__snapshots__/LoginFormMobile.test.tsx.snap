// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LoginFormMobile > matches snapshot 1`] = `
<div>
  <main
    class="_mobileContainer_b69d51"
  >
    <header
      class="_logoSection_b69d51"
    >
      <a
        class="_wrapper_4f3c83"
        href="/"
      >
        <img
          alt="BeerBuddy logo"
          class="_icon_4f3c83"
          src="/teku.svg"
        />
        <h1
          class="_heading_4f3c83"
        >
          BeerBuddy
        </h1>
      </a>
    </header>
    <div
      class="ant-card ant-card-bordered _mobileCard_b69d51 css-dev-only-do-not-override-pr0fja"
      style="background-color: rgb(230, 244, 255);"
    >
      <div
        class="ant-card-body"
      >
        <section>
          <h1
            class="_loginFormHeaderMobile_b69d51"
          >
            Log in
          </h1>
          <form
            autocomplete="off"
            class="ant-form ant-form-vertical css-dev-only-do-not-override-pr0fja _loginFormMobile_b69d51"
            id="basic"
          >
            <div
              class="ant-form-item _loginFormUsername_b69d51 css-dev-only-do-not-override-pr0fja"
            >
              <div
                class="ant-row ant-form-item-row css-dev-only-do-not-override-pr0fja"
              >
                <div
                  class="ant-col ant-form-item-label css-dev-only-do-not-override-pr0fja"
                >
                  <label
                    class="ant-form-item-required"
                    for="basic_username"
                    title="Username"
                  >
                    Username
                  </label>
                </div>
                <div
                  class="ant-col ant-form-item-control css-dev-only-do-not-override-pr0fja"
                >
                  <div
                    class="ant-form-item-control-input"
                  >
                    <div
                      class="ant-form-item-control-input-content"
                    >
                      <input
                        aria-required="true"
                        class="ant-input css-dev-only-do-not-override-pr0fja"
                        id="basic_username"
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="ant-form-item css-dev-only-do-not-override-pr0fja"
            >
              <div
                class="ant-row ant-form-item-row css-dev-only-do-not-override-pr0fja"
              >
                <div
                  class="ant-col ant-form-item-control css-dev-only-do-not-override-pr0fja"
                >
                  <div
                    class="ant-form-item-control-input"
                  >
                    <div
                      class="ant-form-item-control-input-content"
                    >
                      <button
                        class="ant-btn css-dev-only-do-not-override-pr0fja ant-btn-primary _loginButton_b69d51"
                        type="submit"
                      >
                        <span>
                          Submit
                        </span>
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </form>
        </section>
      </div>
    </div>
  </main>
</div>
`;
